audio-rag/rag_code.py

```
@@ -0,0 +1,264 @@
+            indexes = list(self.collection.list_search_indexes(self.VECTOR_INDEX_NAME))
+            if indexes and indexes[0].get("queryable"):
+                return
+            time.sleep(2)
+
+    def ingest_data(self, embeddata):
+        # Clear any previous data for this demo collection so re-uploads don't duplicate.
+        self.collection.delete_many({})
+
+        docs = [
+            {"context": context, self.EMBEDDING_FIELD: embedding}
+            for context, embedding in zip(embeddata.contexts, embeddata.embeddings)
```

Ruff (0.15.17)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)